src/transformStateWithClones.js

```diff
@@ -7,7 +7,26 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newStates = []; // создаем пустой массив
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      currentState = {};
+    } else if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      currentState = { ...currentState };
+      // для removeProperties мы должны создать копию. Перебираем элементы
+
+      for (const key of action.keysToRemove) {
+        delete currentState[key];
+      }
+    }
+    newStates.push(currentState);
```

Comment:
[DONT REPEAT YOURSELF]: Instead of pushing the `currentState` to `newStates` in every case of the if/else if block, do this operation once after the block. This will adhere to the DRY principle and reduce code repetition.